### PR TITLE
Depend on remote artifacts for mantis-shaded instead of local subproject

### PR DESCRIPTION
**Pull request by hmitnflx**
### Context

Currently, we depend on the local copy of jar (subproject) for mantis-shaded dependency. However, this leads to problems with IntelliJ where you have to manually add the mantis-shaded subproject as an explicit library dependency at the root OR deal with a bunch of redlines for these shaded components.

We tried a bunch of alternatives like publishing to local maven or including these shaded dependencies at the terminal level (leaf projects) and they open-up more issues. We decided to host mantis-shaded in a separate repo but we think this could be a better alternative 
### Checklist

- [*] `./gradlew build` compiles code correctly
- [*] Added new tests where applicable
- [*] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable
- [*] Added copyright headers for new files from `CONTRIBUTING.md`

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -68,6 +69,10 @@ subprojects {
         }
     }
 
+    configurations.all {
+        exclude group: 'ch.qos.logback', module: 'logback-classic'
```
Comment (sundargates) [inline build.gradle:73]: you probably want all shaded dependencies here and explicitly filter mantis-shaded here.

File: build.gradle [hunk 2]
```diff
@@ -34,16 +34,17 @@ buildscript {
...
-    versionSlf4j = '1.7.0'
-    versionRxJava = '1.3.8'
-
     junitVersion = '4.11'
     mockitoVersion = '1.9.+'
     slf4jVersion = '1.7.+'
     vavrVersion = '0.9.2'
+
+    versionIoNetty = '4.1.5.Final'
+    versionMantisShaded = '1.3+'
```
Comment (sundargates) [inline build.gradle:43]: should this be `latest.release`?

File: mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/jobmaster/JobMasterService.java [hunk 3]
```diff
@@ -105,8 +104,6 @@ private Measurements handleMetricEvent(final String ev) {
 
         } catch (JsonProcessingException e) {
             logger.error("failed to parse json", e);
-        } catch (IOException e) {
```
Comment (sundargates) [inline mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/jobmaster/JobMasterService.java:108]: Is this a spurious change?

**Review by sundargates — COMMENTED** [review]
Overall looks great! Thanks for fixing this and coming up with the idea! I cannot wait to get this merged.

**Review by codyrioux — APPROVED** [review]

**Review by calvin681 — APPROVED** [review]

File: mantis-discovery-proto/build.gradle [hunk 4]
```diff
@@ -15,7 +15,7 @@
  */
 
 dependencies {
-    api project(path: ':mantis-shaded', configuration: 'shadow')
+    api "io.mantisrx:mantis-shaded:${rootProject.versionMantisShaded}"
```
Comment (sundargates) [inline mantis-discovery-proto/build.gradle:18]: this should be implementation probably?

Comment (hmitnflx) [reply mantis-discovery-proto/build.gradle:18]: I'm doing a like-for-like replacement at the moment, will do this in a separate PR.

File: mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/jobmaster/JobMasterService.java (hunk 3 above)
Comment (hmitnflx) [reply mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/jobmaster/JobMasterService.java:108]: it was a warning during gradle build, will remove and move to a follow-up PR.

File: build.gradle (hunk 2 above)
Comment (hmitnflx) [reply build.gradle:43]: umm, I'll keep it as it is.

File: build.gradle [hunk 5]
```diff
@@ -68,6 +69,20 @@ subprojects {
         }
     }
 
+    configurations.all {
+        if (!"mantis-shaded".equals(project.name)) {
+            exclude group: 'com.fasterxml'
+            exclude group: 'io.vavr.jackson.datatype'
+            exclude group: 'com.google'
```
Comment (sundargates) [inline build.gradle:76]: do you want put the module name too?

**Review by sundargates — APPROVED** [review]
This is an awesome change!

**Pull request merged**